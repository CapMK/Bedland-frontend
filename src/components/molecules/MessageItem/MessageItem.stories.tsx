import type { Meta, StoryObj } from '@storybook/react';
import GlobalStyle from 'theme/globalStyles';
import ThemeProvider from 'theme/ThemeContext';
import Avatar from 'assets/img/Avatar.svg';
import MessageItem from './MessageItem';

const meta = {
  title: 'Molecules/MessageItem',
  component: MessageItem,
  tags: ['autodocs'],
  parameters: {
    backgrounds: { default: 'light' },
  },
  args: { name: 'Jessie Burton', image: Avatar },
  argTypes: {
    name: {
      description: 'Name to display',
    },
    image: {
      table: {
        disable: true,
      },
    },
  },
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        {Story()}
      </ThemeProvider>
    ),
  ],
} satisfies Meta<typeof MessageItem>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {};
