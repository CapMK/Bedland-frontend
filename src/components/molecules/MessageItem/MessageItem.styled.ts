import styled from 'styled-components';
import { PaletteType } from 'theme/theme.types';

export const HoverContainer = styled.div<{ palette: PaletteType }>`
  :hover {
    background: ${({ palette }) => `${palette.sidebarTabsPressed}10`};
  }
  :last-child {
    margin-bottom: 0.75rem;
  }
`;

export const MessageContainer = styled.div<{ palette: PaletteType }>`
  align-items: center;
  border-bottom: 1px solid ${({ palette }) => palette.sidebarTabs};
  display: flex;
  margin-left: 1.125rem;
  margin-right: 1rem;
  padding: 0.875rem 1.938rem 0.875rem 1.438rem;
`;

export const NameWrapper = styled.div`
  flex: 1;
  margin-left: 1rem;
`;
