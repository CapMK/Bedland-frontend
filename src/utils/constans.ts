export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
export const LOGIN_INPUT_WIDTH = 19.875;

// TODO
// BASE_URL points to json server
// Replace with actual API url when BE is ready
export const BASE_URL = 'http://localhost:3001';
