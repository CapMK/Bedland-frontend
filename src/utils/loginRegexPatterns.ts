import { InputVariant, TRegister } from 'components/atoms/Input/Input.types';
import { Getters } from 'types/shared.types';

type LoginRegexProps = Getters<InputVariant, TRegister>;

const loginRegexPattern: LoginRegexProps = {
  login: {
    pattern: /^[a-zA-Z0-9]+$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  password: {
    pattern: /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&*(),<.>/?;:'"]).{8,24}$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  confirmPassword: {
    required: true,
  },
};

export default loginRegexPattern;
// ^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$ <- regex for password that should work.
// /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/,
