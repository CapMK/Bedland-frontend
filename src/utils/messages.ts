const ERROR_MESSAGES = {
  resetPasswordPage: {
    notWithPasswordsRules:
      'Password is incorrect, please try again after reviewing the rules above',

    passwordsDoesNotMatchUp: 'Passwords do not match to each other',
  },
  loginPage: {
    errorMessage: 'Your login or password is incorrect.',
  },
};

export default ERROR_MESSAGES;
