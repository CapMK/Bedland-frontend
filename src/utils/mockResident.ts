const mockResidentList = [
  {
    id: 1,
    version: 0,
    createDate: '2023-02-14T00:00:00',
    updateDate: '2023-02-14T00:00:00',
    login: 'iSutherland',
    password: 'IsABEl',
    name: 'Isabel',
    lastName: 'Sutherland',
    email: 'isabel.sutherland@email.com',
    phoneNumber: '123456789',
    flatId: 1,
    owner: true,
  },
  {
    id: 2,
    version: 0,
    createDate: '2023-02-14T00:00:00',
    updateDate: '2023-02-14T00:00:00',
    login: 'clWeiss',
    password: 'ClWeiss',
    name: 'Claude',
    lastName: 'Weiss',
    email: 'Claude.Weiss@email.com',
    phoneNumber: '123456789',
    flatId: 2,
    owner: false,
  },
];

export default mockResidentList;
