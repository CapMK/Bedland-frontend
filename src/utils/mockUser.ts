import { User } from 'store/reducers/user/user_slice';

export const mockUserCredentials = {
  residentUsername: 'testmember',
  managerUsername: 'testmanager',
  password: '!Test123!',
};

export const loggedInManagerUser = {
  loggedIn: true,
  status: 'success',
  error: undefined,
  userId: '122',
  login: mockUserCredentials.managerUsername,
  token: undefined,
  role: 'manager',
} as User;

export const loggedInResidentUser = {
  loggedIn: true,
  status: 'success',
  error: undefined,
  userId: '122',
  login: mockUserCredentials.residentUsername,
  token: undefined,
  role: 'resident',
} as User;

export const loggedOutManagerUser = {
  loggedIn: false,
  status: 'idle',
  userId: undefined,
  login: undefined,
  token: undefined,
  role: 'manager',
} as User;

export const loggedOutResidentUser = {
  loggedIn: false,
  status: 'idle',
  userId: undefined,
  login: undefined,
  token: undefined,
  role: 'resident',
} as User;
