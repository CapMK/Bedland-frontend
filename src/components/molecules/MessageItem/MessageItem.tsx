import Button from 'components/atoms/Button/Button';
import ProfileImage from 'components/atoms/ProfileImage/ProfileImage';
import Typography from 'components/atoms/Typography/Typography';
import { useContext } from 'react';
import { ThemeContext } from 'theme/ThemeContext';
import * as Styled from './MessageItem.styled';

type MessageItemProps = {
  name: string;
  image: string;
};

const MessageItem = ({ name, image }: MessageItemProps) => {
  const { palette } = useContext(ThemeContext);

  return (
    <Styled.HoverContainer palette={palette}>
      <Styled.MessageContainer palette={palette}>
        <ProfileImage src={image} size={1.5} />
        <Styled.NameWrapper>
          <Typography variant='header4' color='blueDark'>
            {name}
          </Typography>
        </Styled.NameWrapper>
        <Button variant='icon' icon='envelope' />
      </Styled.MessageContainer>
    </Styled.HoverContainer>
  );
};

export default MessageItem;
