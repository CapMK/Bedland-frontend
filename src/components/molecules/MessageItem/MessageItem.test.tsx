import Avatar from 'assets/img/Avatar.svg';
import { cleanup } from '@testing-library/react';
import { renderWithProviders } from 'utils/test.utils';
import MessageItem from './MessageItem';

describe('MessageItem', () => {
  afterEach(cleanup);
  it('renders correctly', () => {
    const messageItem = renderWithProviders(
      <MessageItem name='Jessie Burton' image={Avatar} />,
      {},
    );

    expect(messageItem).toMatchSnapshot();
  });
});
