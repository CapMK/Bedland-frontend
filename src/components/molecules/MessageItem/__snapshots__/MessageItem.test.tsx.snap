// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MessageItem > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  color: #4553DD;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  border-radius: 50%;
  height: 1.5rem;
  width: 1.5rem;
}

.c0:hover {
  background: #919AFF10;
}

.c0:last-child {
  margin-bottom: 0.75rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 1px solid #CACACA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: 1.125rem;
  margin-right: 1rem;
  padding: 0.875rem 1.938rem 0.875rem 1.438rem;
}

.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-left: 1rem;
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <img
            alt="profile"
            class="c2"
            size="1.5"
            src="/src/assets/img/Avatar.svg"
          />
          <div
            class="c3"
          >
            <h4
              style="font-family: Source Sans Pro, sans-serif; font-size: 1rem; font-weight: 600; line-height: 1.4; letter-spacing: 1px; color: rgb(69, 83, 221);"
            >
              Jessie Burton
            </h4>
          </div>
          <button
            class="c4"
          >
            <svg
              data-testid="envelope-icon"
              fill="none"
              height="20"
              viewBox="0 0 20 16"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <mask
                id="mask-envelope"
              />
              <path
                d="M5 5L8.75 8C9.10472 8.28401 9.54559 8.43875 10 8.43875C10.4544 8.43875 10.8953 8.28401 11.25 8L15 5M19 13V3C19 2.46957 18.7893 1.96086 18.4142 1.58579C18.0391 1.21071 17.5304 1 17 1H3C2.46957 1 1.96086 1.21071 1.58579 1.58579C1.21071 1.96086 1 2.46957 1 3V13C1 13.5304 1.21071 14.0391 1.58579 14.4142C1.96086 14.7893 2.46957 15 3 15H17C17.5304 15 18.0391 14.7893 18.4142 14.4142C18.7893 14.0391 19 13.5304 19 13Z"
                stroke="#4553DD"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.67"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  color: #4553DD;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  border-radius: 50%;
  height: 1.5rem;
  width: 1.5rem;
}

.c0:hover {
  background: #919AFF10;
}

.c0:last-child {
  margin-bottom: 0.75rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 1px solid #CACACA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: 1.125rem;
  margin-right: 1rem;
  padding: 0.875rem 1.938rem 0.875rem 1.438rem;
}

.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-left: 1rem;
}

<div>
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <img
          alt="profile"
          class="c2"
          size="1.5"
          src="/src/assets/img/Avatar.svg"
        />
        <div
          class="c3"
        >
          <h4
            style="font-family: Source Sans Pro, sans-serif; font-size: 1rem; font-weight: 600; line-height: 1.4; letter-spacing: 1px; color: rgb(69, 83, 221);"
          >
            Jessie Burton
          </h4>
        </div>
        <button
          class="c4"
        >
          <svg
            data-testid="envelope-icon"
            fill="none"
            height="20"
            viewBox="0 0 20 16"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <mask
              id="mask-envelope"
            />
            <path
              d="M5 5L8.75 8C9.10472 8.28401 9.54559 8.43875 10 8.43875C10.4544 8.43875 10.8953 8.28401 11.25 8L15 5M19 13V3C19 2.46957 18.7893 1.96086 18.4142 1.58579C18.0391 1.21071 17.5304 1 17 1H3C2.46957 1 1.96086 1.21071 1.58579 1.58579C1.21071 1.96086 1 2.46957 1 3V13C1 13.5304 1.21071 14.0391 1.58579 14.4142C1.96086 14.7893 2.46957 15 3 15H17C17.5304 15 18.0391 14.7893 18.4142 14.4142C18.7893 14.0391 19 13.5304 19 13Z"
              stroke="#4553DD"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.67"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
